R/fitNullModel.R

```diff
@@ -193,3 +221,83 @@ nullModelSmall <- function(null.model) {
 isNullModelSmall <- function(null.model) {
     is.null(null.model$cholSigmaInv)
 }
+
+
+.modelString <- function(outcome, covars = NULL, random = NULL, group.var = NULL, inverse_normal = FALSE) {
+    model.outcome <- .modelOutcomeString(outcome, inverse_normal = inverse_normal)
+    model.covars <- if (is.null(covars)) NULL else .modelCovarString(covars)
+    model.random <- if (is.null(random)) NULL else paste(paste0("(1|", random, ")"), collapse=" + ")
+    model.var <- if (is.null(group.var)) NULL else paste0("var(", group.var, ")")
+    model.string <- paste(c(model.covars, model.random, model.var), collapse=" + ")
+    paste(model.outcome, model.string, sep=" ~ ")
+}
+
+.modelCovarString <- function(covars) {
+  paste(covars, collapse=" + ")
+}
+
+.modelOutcomeString <- function(outcome, inverse_normal = FALSE) {
+  if (inverse_normal) {
+    return(sprintf("rankInvNorm(resid(%s))", outcome))
+  } else {
+    return(outcome)
+  }
+}
+
+.setCovMatNames <- function(cov.mat) {
+  if (!is.list(cov.mat)) {
+    cov.mat <- list(cov.mat)
+  }
+  # What about the case where one element of the matrix has names, and the others don't?
```

Comment:
Can we assign `LETTERS` to ones that have no name? Or should we just always require names?

Reply:
I decided against this because if I had a list like:
```
l <- list(
  A = matrix(runif(4), 2,2),
  matrix(runif(4), 2, 2)
)
```
It would assign `A` to the second list element unless we then also checked that no names were duplicated (which is probably a check we should do). I think it's better to have them all named by the user or non named by the user and then named by the code.

Reply:
Makes sense.